### PR TITLE
Refactor makefile script

### PR DESCRIPTION
**Pull request by SuperQ**
* Fix shellcheck issues
* Fix some shell style inconsistencies.
* Split out some functions.

Signed-off-by: Ben Kochie <superq@gmail.com>

### CODE REVIEW

File: scripts/sync_makefiles.sh [hunk 1]
```diff
@@ -15,59 +15,85 @@ orgs="prometheus prometheus-community"
...
-cd "$(git rev-parse --show-cdup)"
+cd "$(git rev-parse --show-cdup)" || exit 1
 
 source_makefile="$(pwd)/Makefile.common"
 source_checksum="$(sha256sum Makefile.common | cut -d' ' -f1)"
 
-tmp_dir=$(mktemp -d)
-trap "rm -rf ${tmp_dir}" EXIT
+tmp_dir="$(mktemp -d)"
+trap 'rm -rf "${tmp_dir}"' EXIT
```
Comment (roidelapluie) [inline scripts/sync_makefiles.sh:29]: ${tmp_dir:?}

File: scripts/sync_makefiles.sh [hunk 2]
```diff
@@ -15,59 +15,85 @@ orgs="prometheus prometheus-community"
...
+    "https://${GITHUB_TOKEN}:@github.com/${1}" \
+    --set-upstream "${branch}" 1>/dev/null 2>&1
+}
+
+post_template='{"title":"%s","base":"master","head":"%s","body":"%s"}'
+post_json="$(printf "${post_template}" "${pr_title}" "${branch}" "${pr_msg}")"
+post_pull_request() {
+  curl --show-error --silent --fail \
+    -u "${git_user}:${GITHUB_TOKEN}" \
+    -X POST \
```
Comment (roidelapluie) [inline scripts/sync_makefiles.sh:50]: I'd suggest we remove that line: https://daniel.haxx.se/blog/2015/09/11/unnecessary-use-of-curl-x/

File: scripts/sync_makefiles.sh (hunk 1 above)
Comment (SuperQ) [reply scripts/sync_makefiles.sh:29]: I don't think this is necessary as the whole script is `set -u`.

File: scripts/sync_makefiles.sh (hunk 2 above)
Comment (SuperQ) [reply scripts/sync_makefiles.sh:50]: Hah, yes, that's fine with me.

**Comment by SuperQ** [comment]
@roidelapluie Any other comments?

**Comment by roidelapluie** [comment]
LGTM :)

**Pull request merged**